src/pytest/test.py

```diff
@@ -2967,6 +2967,34 @@ def testUnseportedSortableTypeErrorOnTags(env):
     env.expect('HGETALL doc1').equal(['f1', 'foo1', 'f2', '2', 'f3', 'foo2', 'f4', 'foo2'])
     env.expect('FT.SEARCH idx *').equal([1L, 'doc1', ['f1', 'foo1', 'f2', '2', 'f3', 'foo2', 'f4', 'foo2']])
 
+def testIssue1063(env):
+    env.cmd('FT.CREATE idx SCHEMA txt1 TEXT txt2 TEXT')
+
+    env.cmd('FT.ADD idx doc1 1.0 FIELDS txt1 10')
+    env.expect('FT.GET idx doc1').equal(['txt1', '10'])
+
+    env.expect('FT.ADD idx doc1 1.0 REPLACE PARTIAL if isnull(@txt1)||@txt1!=10 FIELDS txt1 10').equal('NOADD')
+    env.expect('FT.ADD idx doc1 1.0 REPLACE PARTIAL if isnull(@txt1)||@txt1==10 FIELDS txt1 10').equal('OK')
+    env.expect('FT.ADD idx doc1 1.0 REPLACE PARTIAL if isnull(@txt2)||@txt1!=10 FIELDS txt1 10').equal('OK')
+    env.expect('FT.ADD idx doc1 1.0 REPLACE PARTIAL if isnull(@txt2)||@txt1==10 FIELDS txt1 10').equal('OK')
+
+    env.expect('FT.ADD idx doc1 1.0 REPLACE PARTIAL if @txt1!=10||isnull(@txt1) FIELDS txt1 10').equal('NOADD')
+    env.expect('FT.ADD idx doc1 1.0 REPLACE PARTIAL if @txt1==10||isnull(@txt1) FIELDS txt1 10').equal('OK')
+    env.expect('FT.ADD idx doc1 1.0 REPLACE PARTIAL if @txt1!=10||isnull(@txt2) FIELDS txt1 10').equal('OK')
+    env.expect('FT.ADD idx doc1 1.0 REPLACE PARTIAL if @txt1==10||isnull(@txt2) FIELDS txt1 10').equal('OK')
+
+    env.expect('FT.ADD idx doc1 1.0 REPLACE PARTIAL if isnull(@txt1)&&@txt1!=10 FIELDS txt1 10').equal('NOADD')
+    env.expect('FT.ADD idx doc1 1.0 REPLACE PARTIAL if isnull(@txt1)&&@txt1==10 FIELDS txt1 10').equal('NOADD')
+    env.expect('FT.ADD idx doc1 1.0 REPLACE PARTIAL if isnull(@txt2)&&@txt1!=10 FIELDS txt1 10').equal('NOADD')
+    env.expect('FT.ADD idx doc1 1.0 REPLACE PARTIAL if isnull(@txt2)&&@txt1==10 FIELDS txt1 10').equal('OK')
+
+    env.expect('FT.ADD idx doc1 1.0 REPLACE PARTIAL if @txt1!=10&&isnull(@txt1) FIELDS txt1 10').equal('NOADD')
+    env.expect('FT.ADD idx doc1 1.0 REPLACE PARTIAL if @txt1==10&&isnull(@txt1) FIELDS txt1 10').equal('NOADD')
+    env.expect('FT.ADD idx doc1 1.0 REPLACE PARTIAL if @txt1!=10&&isnull(@txt2) FIELDS txt1 10').equal('NOADD')
+    env.expect('FT.ADD idx doc1 1.0 REPLACE PARTIAL if @txt1==10&&isnull(@txt2) FIELDS txt1 10').equal('OK')
+
+    env.expect('FT.ADD idx doc1 1.0 REPLACE PARTIAL if isnull(to_number(@txt1)) FIELDS txt1 10').equal('NOADD')
+    env.expect('FT.ADD idx doc1 1.0 REPLACE PARTIAL if isnull(to_number(@txt2)) FIELDS txt1 10').equal('NOADD')
```

Comment:
This should be error and not noadd because t2 is not exists so its to_num should raise error